src/lightning_lite/plugins/environments/slurm_environment.py

```diff
@@ -141,3 +149,36 @@ def resolve_root_node_address(nodes: str) -> str:
         nodes = re.sub(r"\[(.*?)[,-].*\]", "\\1", nodes)  # Take the first node of every node range
         nodes = re.sub(r"\[(.*?)\]", "\\1", nodes)  # handle special case where node range is single number
         return nodes.split(" ")[0].split(",")[0]
+
+    @staticmethod
+    def _validate_srun_used() -> None:
+        """Checks if the `srun` command is available and used.
+
+        Parallel jobs (multi-GPU, multi-node) in SLURM are launched by prepending `srun` in front of the Python command.
+        Not doing so will result in processes hanging, which is a frequent user error. Lightning will emit a warning if
+        `srun` is found but not used.
+        """
+        srun_exists = subprocess.call(["which", "srun"]) == 0
+        srun_used = "SLURM_NTASKS" in os.environ
+        hint = " ".join(["srun", os.path.basename(sys.executable), *sys.argv])[:64]
```

Comment:
why do we limit this to 64 chars?

Reply:
In case the python command is long (maybe many hyperparameters) we truncate it so the warning message remains readable. 64 is an arbitrary choice and the author likes powers of 2.